### DISCUSSION TITLE
What can I use in XAML code to create a skiaSharp render context?

### DISCUSSION THREAD

**Discussion by LyndonAlcock** [AvaloniaUI/Avalonia#7092] [Q&A]
I'm looking to develop my skills in software but i'm a massive beginner, i need a way to quickly set up a SkiaSharp canvas in order to use is this possible or should i migrate my app to javascript?

**Comment by maxkatz6** [comment]
You can take any control and override Render method. 
For example:
```csharp
public class MyControl : Control
{
   public override void Render(DrawingContext context)
   {
       if (context.PlatformImpl is ISkiaDrawingContextImpl skiaContext)
       {
            var canvas = skiaContext.SkCanvas;
            var surface = skiaContext.SkSurface;
       }
       else
       {
            throw new NotSupportedException("Only Skia rendering backend is supported");
       }
   }
}
```

**Comment by maxkatz6** [comment]
Or if you want a bit more low lever implementation, you can use ICustomDrawOperation, which will be invoked on rendering thread (not UI thread as another option).
You can find example here https://github.com/AvaloniaUI/Avalonia/blob/7842883961d094e08e9def7f30cf32fd573179c7/samples/RenderDemo/Pages/CustomSkiaPage.cs#L21
